### PR TITLE
Staging Push 1021 - PIV Authentication Troubleshooting and Identifiers

### LINKED ISSUE #82: Error logging to enable on domains for troubleshooting
Add information, configurations and scripts for which error logs to enable on network domains for enterprise engineers to troubleshoot. 

List of the common errors encountered mapped to primary and secondary _troubleshooting_ specific to federal agencies

### LINKED ISSUE #84: Which platforms break when not using principal name?
We need some feedback here to understand any repercussions with not using Principal Name mapping for network.  

Primarily for common implementations and office automation tools (cloud email providers, common federation services).

### PR DESCRIPTION
**Pull request by maxwellfunk**
Closes #82, Closes #84 

All information was added to the PIV playbook.  

Added RFC822 in SAN reference to the PIV identifiers page per input from Todd on issue #84, and added much more information to network authentication troubleshooting to include CAPI2 screenshots on issue #82.

### CODE REVIEW

**Review by JBPayne007 — APPROVED** [review]

File: _data/navigation.yml [hunk 1]
```diff
@@ -89,6 +89,8 @@ pivnetwork:
     href: /piv/network/localca/
   - text: Authentication Assurance
     href: /piv/network/auth/
+  - text: Troubleshooting PIV Auth Issues
```
Comment (idmken) [inline _data/navigation.yml:92]: I updated this. using "common" made it seem like it was related to COMMON.

File: _piv/network/13_common_issues.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  - text: Account Linking
+    href: '#account-linking'
+  - text: Helpful Tools
+    href: '#helpful-tools'
+---
+
+## Trust Stores
+PKI based mutual authentication issues often arrive when a host (e.g., a domain contoller) and client (e.g., a workstation) are not capable of establishing trust.  Given the path discovery and validation (PDVAL) processes take place within both machines, potential errors due to multiple scenarios with either machine.
+
+   - Domain Controller (Host)
```
Comment (idmken) [inline _piv/network/13_common_issues.md]: You've outlined two major types of troubleshooting steps. I'd expect to see steps related to teach major area below to investigate. I don't see this. @rsherwood-gsa

File: _piv/network/13_common_issues.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      - Domain Controller unable to validate PIV certificate
+      - NTAuth does not contain PIV root CA certificate
+      - NTAuth does not contain PIV intermediate or issuing CA certificate
+      - Domain Controller unable to validate its own certificate (expiration) 
+   - User Workstation (Client)
+      - User workstation unable to validate PIV certificate 
+      - NTAuth does not contain DC root CA certificate
+      - NTAuth does not contain DC intermediate or issuing CA certificate
+
+In these instances you may want to view the other end-entity certificate on the opposite machine and check the certifiation path to determine if there are errors in path validation.  If errors are encountered, you may have to modify each trust store (Root and Intermeidate CAs) to include the needed CAs from trust path.
```
Comment (idmken) [inline _piv/network/13_common_issues.md]: This is confusing. "view the other certs on the opposite machine"? If the audience for this is someone troubleshooting an authentication error. I'd expect to see steps. step 1 check here. step 2 check here. Or a list of common errors and what error they relate to. For example, a "smart card authentication is not supported" error message related to error code xx in the event viewer and related to domain authentication or a GPO, etc.

File: _piv/network/13_common_issues.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+## Network Misconfigurations
```
Comment (idmken) [inline _piv/network/13_common_issues.md]: Make this clearer that "sometimes network rules restrict access to PKI necessary locations for revocation checking". Link to COmmon playbook or network tuning on how to check your machine can access the appropriate URL.

File: _piv/network/13_common_issues.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+Due to required security controls many networks appliances are equiped with stringent IP whitelists that may be blocking publicly avaialble PKI resources such as Authority Information Access (AIA) points, Credential Revocation List (CRL) distrobution points, or Online Certificate Status Protocol (OCSP) responder addresses.  Common issues related to network limitations may include:
+
+   - Revocation data such as CRL or OCSP traffic being blocked
+   - Path buidling data such as SIA or AIA traffic being blocked
+   - [Network Tuning](/../network/tuning){:target="_blank"} related misconfiguration such as CRL download time limits
+
+You may need to coordinate with your network or active directory administrators to allow traffic to/from certain IP addresses or make changes to relevant group policies as needed.
+
+## Account Linking
```
Comment (idmken) [inline _piv/network/13_common_issues.md]: Plain language this . "the AD account name must match the UPN in the PIV card. If it doesn't you may see xx error with error code xx in the event log on the workstation and domain controller. Go this xx log in the event viewer and look for error code xx. Reference our account linking page for instructions on account linking or this technet article on xx".

File: _piv/network/13_common_issues.md (hunk 2 above)
Comment (idmken) [reply _piv/network/13_common_issues.md]: Based on this list, I would expect to see below sections on.
1. domain controller or workstation unable to validate PIV certificate (network misconfiguration)
2. NTAuth misconfigurations (trust stores?)
3. domain controller certificate validation (?)

Should add account linking under domain controller issues to match the below.

File: _piv/network/13_common_issues.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+You may need to coordinate with your network or active directory administrators to allow traffic to/from certain IP addresses or make changes to relevant group policies as needed.
+
+## Account Linking 
+Sometimes, even though hosts and clients might be able to complete mutual PKI authentication as designed, other downstream issues may arise that can prevent successful PIV use for network access.  Primarily these issues may lie with unique identification of the credential and its correlation to a network account.  Some of specific issues with unique identification may include:
+
+  - AltSecID or User Principal Name (UPN) imporperly formatted
+  - User selects improper certificate during authentication
+
+You can visit our [network account](/../network/account){:target="_blank"} page for additional guidance in these situations.
+
```
Comment (idmken) [inline _piv/network/13_common_issues.md]: expecting to see remediation steps related to domain controller certificate errors.
- typical errors either in the GUI or event view.
- Playbook or technet links.

File: _piv/network/13_common_issues.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+## Helpful Tools
+
+Microsoft Windows based systems have the ability to record logs associated with PKI and X.509 based events, called CryptoAPI v2.0 or CAPI2.  This utility can be very helpful when enabled on a domain controller host or a client machine in identifying issues with PKI processes.
```
Comment (idmken) [inline _piv/network/13_common_issues.md]: What are the different errors an admin would see between the domain controller and the workstation and why would someone look at one vs the other?

**Review by idmken — CHANGES_REQUESTED** [review]
@rsherwood-gsa @maxwellfunk Audience for this seems like an admin who is trying to troubleshoot PIV authentication errors. Please update based on my comments and do an overall plain language review.

File: _piv/5_piv_identifiers.md [hunk 8]
```diff
@@ -24,7 +24,8 @@ The table below outlines identifiers available in the PIV Authentication certifi
...
 | Card Universal Unique Identifier (UUID)      |   Unique for every person and credential; Value changes when a user receives a new, replaced or updated PIV credential; Card UUID value is only required to be present for new or replacement PIV credentials issued after August 2014; May also commonly be referred to as the Global Unique Identifier (GUID). |
-| User Principal Name in _Subject Alternate Name_   |  Not required to be included in all PIV Authentication certificates; Not recommended for use as an identifier to achieve full interoperability for networks or applications; Commonly used for enterprise smart card logon / network authentication in _legacy_ implementations. |
+| User Principal Name in _Subject Alternative Name_   |  Unique within a given domain; Not required to be included in all PIV Authentication certificates; Not recommended for use as an identifier to achieve full interoperability for networks or applications; Commonly used for enterprise smart card logon / network authentication in _legacy_ implementations. |
```
Comment (idmken) [inline _piv/5_piv_identifiers.md:27]: If this addition to to answer the question "what breaks without UPN", it doesn't answer that.

**Review by idmken — CHANGES_REQUESTED** [review]

File: _piv/5_piv_identifiers.md (hunk 8 above)
Comment (maxwellfunk) [reply _piv/5_piv_identifiers.md:27]: Given Todd's comments on the issue, I think the scope of the issue from 2016 has changed perspective.  Most web-applications are looking for an identifier that might be from an external domain that also provides user validation capability.  As a result, the preference for RFC822 email would seem to be more practical guidance for modern issues.  A UPN of 1234567890@mail.mil (e.g. a DoD EDIPI based UPN may not even be used).  Additionally, we received no feedback on what web platforms break in any instance.

File: _piv/network/13_common_issues.md (hunk 7 above)
Comment (maxwellfunk) [reply _piv/network/13_common_issues.md]: > @rsherwood-gsa @maxwellfunk Audience for this seems like an admin who is trying to troubleshoot PIV authentication errors. Please update based on my comments and do an overall plain language review.

Due to the potential for the client to be unable to validate a DC cert in a mutual authentication scenario, I think the intended audience is actually broader.

**Comment by JillTunick** [comment]
@maxwellfunk and @rsherwood-gsa, after you've made the content additions/changes that Ken requested, please tag me so I can do plain language review and editing on the page. Thanks in advance.

**Comment by rsherwood-gsa** [comment]
@idmken, based on our chat on Friday, I have tried to do some restructuring of piv-auth-issues to have a page flow - there is in introductory section and then some sub-bullets that organize the material. A lot of work to do in the details to address your comments, but do you like the structure I have proposed?

**Pull request closed without merging**